app/grandchallenge/participants/views.py

```diff
@@ -94,8 +98,48 @@ def get_queryset(self):
         queryset = queryset.filter(
             Q(challenge=self.request.challenge)
         ).select_related("user__user_profile", "user__verification")
+
         return queryset
 
+    def get_context_data(self, **kwargs) -> dict[str, Any]:
+        context_data = super().get_context_data(**kwargs)
+
+        questions = self._get_registration_questions()
+        context_data["registration_questions"] = questions
+
+        answers = self._get_registration_questions_answers(questions=questions)
+
+        answer_lookup = defaultdict(dict)
+        for answer in answers:
+            answer_lookup[answer.registration_request.pk][
+                answer.question.pk
+            ] = answer
+
+        context_data["registration_questions_answer_lookup"] = answer_lookup
+
+        return context_data
+
+    def _get_registration_questions(self):
+        return filter_by_permission(
+            queryset=self.request.challenge.registration_questions,
+            user=self.request.user,
+            codename="view_registrationquestion",
+            accept_user_perms=False,
+        )
+
+    def _get_registration_questions_answers(self, questions):
+        queryset = RegistrationQuestionAnswer.objects.filter(
+            registration_request__in=self.object_list,
+            question__in=questions,
+        ).select_related("registration_request", "question")
+
+        return filter_by_permission(
+            queryset=queryset,
+            user=self.request.user,
+            codename="view_registrationquestionanswer",
+            accept_user_perms=False,
+        )
```

Comment:
This feels like a little bit of overkill. If the user can see the request they should be able to see the questions and answers anyway right?